pallets/runtime/src/multisig.rs

```diff
@@ -272,8 +265,50 @@ decl_module! {
                 <NumberOfSigners<T>>::get(&sender) >= sigs_required,
                 Error::<T>::NotEnoughSigners
             );
-            <MultiSigSignsRequired<T>>::insert(&sender, &sigs_required);
-            Self::deposit_event(RawEvent::MultiSigSignaturesRequiredChanged(sender, sigs_required));
+            Self::unsafe_change_sigs_required(sender, sigs_required);
+            Ok(())
+        }
+
+        /// This function allows to replace all existing signers of the given multisig & also change no. of signature required
+        /// NOTE - Once this function get executed no other function of the multisig is allowed to execute until unless
+        /// potential signers accept the authorization and there count should be greater than or equal to the signature required
```

Comment:
Some speculative remarks about this comment.

Maybe there should be a flag for this in `decl_storage` and checks whether this flag is not set in every dispatchable function in `multisig` to guarantee that? That sounds a bit dubious to me though because how would we unset such a flag once all authorisations are accepted?

On another hand - just speculating - you could remove the votes from old, removed signers from `Votes` and subtract the number of those votes from `TxApprovals`. But that would be a lot of redundant work since both `Votes` and `TxApprovals` are append-only. That is, unless there is a list of currently open proposals, in which case only those open proposals are modified in `Votes` and `TxApprovals`.

Having said that, I actually can't see any problem in new signers starting to cast their votes before all the authorisations are accepted. That assumes we keep trust in the votes of the removed validators. So I think the note about the authorisations could be removed. However the note about the number of signatures means that

* you either need to add a check that `sigs_required` is greater or equal to the number of currently required signatures (which will ensure the invariant you require in the note), and later `sender` could lower it with `change_sigs_required`,

* or you can allow existing proposals getting approved with less signers should the decision of the `sender` say so.

So, again, I think you could either remove the note about the number of signers or add some checks.